src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -367,7 +365,7 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
                 }
             }
 
-            CheckModifiers(location, diagnostics);
+            CheckModifiers((syntax.Body ?? (SyntaxNode)syntax.ExpressionBody) != null, location, diagnostics);
```

Comment:
>(syntax.Body ?? (SyntaxNode)syntax.ExpressionBody) [](start = 27, length = 50)

This pattern occurs multiple time. Should we factor it into the syntax or an extension?
Then `(declaration = GetSyntax()).Body == null && declaration.ExpressionBody == null` below (line 595) also simplifies to `(declaration = GetSyntax()).BestBody == null`